### ISSUE TITLE
Update lua-Harness suite

### ISSUE THREAD

**Issue by igormunkin** [tarantool/tarantool#5970]
We have incorporated [lua-Harness](https://framagit.org/fperrad/lua-Harness) suite into our [luajit](https://github.com/tarantool/luajit) repo within tarantool/luajit@347b8bf. Huge thanks to @fperrad, our experience has been considered and the issues we faced are finally fixed. There is a table below, mapping our hacks to the corresponding patches in [lua-Harness trunk](https://framagit.org/fperrad/lua-Harness/-/commits/master/):

| Problem | Temporary solution | Patch in the trunk |
| - | - | - |
| resolving program name | tarantool/luajit@8376885 | [fperrad/lua-Harness@1be25a8](https://framagit.org/fperrad/lua-Harness/-/commit/1be25a8e6937ae3e655c732cccec3952ec290ee3) |
| out-of-source testing | tarantool/luajit@d11c5bb | [fperrad/lua-Harness@60da289](https://framagit.org/fperrad/lua-Harness/-/commit/60da289a604bd517edde797c4a06d98255145589) |
| CI environment | tarantool/luajit@45ed138 | [fperrad/lua-Harness@6c2aa87](https://framagit.org/fperrad/lua-Harness/-/commit/6c2aa8798f818755cc0de920a244bd07fd1296ff) |
| TAP modules name collision | tarantool/tarantool@a0ade20 | [fperrad/lua-Harness@8041c45](https://framagit.org/fperrad/lua-Harness/-/commit/8041c45ee26501739b512856acfc20ffeb292430) |
| `314-regex.t` bug | tarantool/luajit@789820a | [fperrad/lua-Harness@24a570c](https://framagit.org/fperrad/lua-Harness/-/commit/24a570c1bd4e8ec04dc0a0d013fea0958a0722a5) |
| CLI tests | tarantool/luajit@d4e12d7 | [fperrad/lua-Harness@1da5b1b](https://framagit.org/fperrad/lua-Harness/-/commit/1da5b1b386c748024cf0a33baacdb40740b10f91) |

Furthermore, there is a separate [Tarantool profile](https://framagit.org/fperrad/lua-Harness/-/blob/master/test_lua/profile_tarantool.lua), already merged to the master branch, that need to be tested using our CI machinery. This changeset resolves the remaining problems we faced while adopting the suite.

Considering everything above I propose to bump lua-Harness suite up to [673c7869](https://framagit.org/fperrad/lua-Harness/-/commit/673c7869e4c697e8f5a3cc9ae9d15eaf02d2b826) and test Tarantool profile.

**Comment by igormunkin** [comment]
After applying [b7b1a9a2](https://framagit.org/fperrad/lua-Harness/-/commit/b7b1a9a2f35d7393624e9eb8e59010b1fe3b44be), we faced two new issues:

* `241-standalone.t`: another problem with damn MacOS, because they can't implement interfaces working the same way these functions work on Linux. As a result, [the introduced assertion](https://framagit.org/fperrad/lua-Harness/-/commit/1da5b1b386c748024cf0a33baacdb40740b10f91#d63d55ea4e25efeb575db8997e8df66ce5856d0c_188_204) for Tarantool error message fails, since `getopt_long(3)` yields the error message without single quotes wrapping the flag. Consider the following difference:
  + Linux:
  ```
  $ uname -mrs
  Linux 4.19.66-gentoo x86_64
  $ ./src/tarantool -e
  ./src/tarantool: option requires an argument -- 'e'
  ```
  + MacOS:
  ```
  $ uname -mrs            
  Darwin 20.5.0 arm64
  $ ./src/tarantool -e
  tarantool: option requires an argument -- e
  ```
  **Affected pipelines**: [`osx_10_15`](https://github.com/tarantool/tarantool/runs/3061487552), [`osx_10_15_lto`](https://github.com/tarantool/tarantool/runs/3061488380), [`osx_11_0`](https://github.com/tarantool/tarantool/runs/3061487553), [`osx_arm64_11_2`](https://github.com/tarantool/tarantool/runs/3061488150), [`osx_debug_arm64_11_2`](https://github.com/tarantool/tarantool/runs/3061487566).
* `303-package.t`: [`package.searchpath` assertion](https://framagit.org/fperrad/lua-Harness/-/commit/074fefa51da1276078f735fa8fb568963fe541f1?page=2#42eea497bab20de4ccbf2514bbe55691de62c289_135_133) has been changed in scope of [conversion to Test.Assertion](https://framagit.org/fperrad/lua-Harness/-/commit/074fefa51da1276078f735fa8fb568963fe541f1). As a result out of this change out-of-source testing is broken, but the previous assertion using `package.path` as a second argument worked fine.
  **Affected pipelines**: [`out_of_source`](https://github.com/tarantool/tarantool/runs/3061488374), [`static_build_cmake_linux`](https://github.com/tarantool/tarantool/runs/3061488119).
  
I've also reported the faced issue to @fperrad via email.

**Comment by igormunkin** [comment]
Thanks to @fperrad: both issues are fixed in the [trunk](https://framagit.org/fperrad/lua-Harness/-/commits/master/test_lua).
* [Fix for `241-standalone.t`](https://framagit.org/fperrad/lua-Harness/-/commit/a0532c5)
* [Fix for `303-package.t`](https://framagit.org/fperrad/lua-Harness/-/commit/c4451fe)

However, when `241-standalone.t` has been enabled back to our CI, Tarantool interactive mode misbehaviour has been found on FreeBSD (for more info see #6231). Hence, I propose to mask [this particular assertion](https://framagit.org/fperrad/lua-Harness/-/blob/master/test_lua/241-standalone.t#L115-130) for FreeBSD via separate commit, until the mentioned issue is resolved.

**Comment by igormunkin** [comment]
The issue is resolved in scope of tarantool/luajit@e7f7016. The submodule was updated in tarantool in 2.9.0-185-g722566572 (7225665), 2.8.1-135-g901b626cc (901b626), 2.7.2-126-gb6dd2d6d3 (b6dd2d6), 1.10.10-59-g3f70f2736 (3f70f27).

Tarantool profile integration is moved to #6279.
